### DISCUSSION TITLE
passing attributes to events

### DISCUSSION THREAD

**Discussion by acivitillo** [idom-team/idom#633] [Question] [answered]
Is it possible to pass on additional attributes to events?

```python

def handle_click(event):
  print(event["target"]["value"]) #<- returns KeyValue Error

html.a({"onClick":handle_click", "value":"somevalue"})
````

The `value` attribute seems to work only with certain elements like `html.input`, but not for others like `html.a`.

**Comment by rmorshea** [comment, marked as answer]
IDOM serializes event information [here](https://github.com/idom-team/idom/blob/main/src/client/packages/idom-client-react/src/event-to-object.js). Then, for attributes of an event target that get serialized, those are handled [here](https://github.com/idom-team/idom/blob/37a5be27173a3a82df11650126da3f689a074192/src/client/packages/idom-client-react/src/event-to-object.js#L59-L71). In the case of the `value` attribute, it's only present for the following element types:

- BUTTON
- INPUT
- OPTION
- LI
- METER
- PROGRESS
- PARAM
- SELECT
- TEXTAREA

If there's any missing event or event target information that should be included I'd be happy to take a contribution to fix that.

**Reply by Archmonger** [reply]
Are `data-*` attributes serialized? If not then they probably always should be.

**Reply by rmorshea** [reply]
It does not. That should be added though.

**Reply by rmorshea** [reply]
Should be pretty easy to [grab it](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset)

**Reply by acivitillo** [reply]
OK, switched to the `button` element for now. @Archmonger the `data-*` is exactly what I was referring to now that you make me think about it.

**Reply by Archmonger** [reply]
I've created an issue for it.

- https://github.com/idom-team/idom/issues/634

Developing that is lower down on the priority list though, since there are workarounds for this.

If you want to see this feature sooner rather than later you could contribute a PR!

**Reply by acivitillo** [reply]
I understand the priority is not high, with some styling using button instead of anchor works quite ok as a workaround. 

I am not sure I am able to contribute just yet, only docs, for now, I am still getting familiar with IDOM by writing a demo app.
